updates/P7.txt

```diff
@@ -0,0 +1,320 @@
+effective stakes of bakers and delegators are sampled in the first block
+of the last epoch before each payday, with the sampled stakes being used
+from the upcoming payday. Although the change is recorded on the account
+immediately, it will be between 1 epoch and 1 payday plus 1 epoch until
+the change affects the proof-of-stake lottery used for consensus.)
+Stake in the cooldown queue is considered to be inactive stake, and, as
+such, does not contribute to the lottery calculation. However, the
+inactive stake still cannot be spent, although it is possible to transfer
+it back to active stake.
+
+When stake is removed from a validator or delegator, it enters cooldown
+at the point that the removed stake is no longer part of the effective
+stake for lottery calculations. Specifically, when the stake is removed,
+it first enters a "pre-pre-cooldown" state. When the stake snapshot is
+taken for next payday (in the first block of the epoch before the payday)
+it is moved from "pre-pre-cooldown" to "pre-cooldown". At the subsequent
+payday, it is moved from "pre-cooldown" into "cooldown", and the cooldown
+expiry time is set based on the cooldown time chain parameter. In
+particular, the cooldown officially starts at this payday. At the first
+payday at or after the cooldown expiry time, the stake finally exits
+"cooldown" and becomes available to spend. An account's active stake
+can be reduced while there is still inactive stake in cooldown, leading
+to multiple amounts in cooldown (or pre-cooldown or pre-pre-cooldown)
+that expire at different times.
+
+If the active stake on an account is increased, the increase is first
+transferred from the inactive stake in preference to taking from the
+unstaked balance. Specifically, it is taken first from "pre-pre-cooldown",
+then "pre-cooldown", and then from the "cooldown" stakes in decreasing
+order of expiry time. An account can also switch freely between being
+a validator and being a delegator without its stake entering cooldown.
+(Prior to protocol version 7, it was necessary to remove the validator
+or delegator on an account, waiting out the cooldown period, before
+instating a delegator or validator.) As a consequence of this, from
+protocol version 7, no distinction is drawn between the cooldown time
+for validators and for delegators: stake entering cooldown expires
+after the minimum of both times.
+
+The purpose of these changes is to maximise the flexibility of staking,
+while still ensuring that any stake must still be locked for the
+cooldown period before it can be spent.
+
+                                * * *
+
+Since the initial protocol version, the Concordium block chain has
+supported shielded transfers (also known as encrypted transfers).
+This mechanism allows an account to "shield" part of its balance, 
+cryptographically encrypting it so that the amount cannot be inspected
+except by the owner of the account (using its secret encryption key).
```

Comment:
**Minor detail:** Maybe not the most important detail here, but an authority could decrypt these by convincing a Identity disclosure authority, right?